param.go

```diff
@@ -146,36 +146,26 @@ func (pl paramList) Build(containerStore) (reflect.Value, error) {
 // to the underlying constructor.
 func (pl paramList) BuildList(c containerStore) ([]reflect.Value, error) {
 	args := make([]reflect.Value, len(pl.Params))
-	argsBuilt := make([]bool, len(pl.Params))
 	allContainers := c.getStoresFromRoot()
 	for i, p := range pl.Params {
-		var err error
-		var arg reflect.Value
 		// iterate through the tree path of scopes.
 		for _, c := range allContainers {
```

Comment:
Just confirming: We do this in parent->child order. Is the reason for this that we want the value to be instantiated in the parent and shared across all descendants if possible?